sdk/web-pubsub/web-pubsub/src/hubClient.ts

```diff
@@ -286,7 +287,7 @@ export class WebPubSubServiceClient {
       ...this.clientOptions,
       ...{
         apiVersion: this.apiVersion,
-        loggingOptions: {
+        loggingOptions: this.clientOptions?.loggingOptions ?? {
```

Comment:
I feel like we shouldn't allow the consumer to override the `logger` itself, nor should we not pass `logger.info` here just because they're specified some options.

Is the specific ask to log additional headers or both additional headers and query params? Assuming it's only headers I would extend your client options to take `additionalAllowedHeaderNames` and pass this explicitly along with `loggingOptions` instead of exposing the entire internal option bag to your consumer.

Reply:
makes total sense to me, updated